IngressMaxFields.user.js

```diff
@@ -69,66 +69,76 @@ function wrapper() {
 		});
 		return c;
 	}
+	self.inBounds = function(portal) {
+		if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
+			return self.portalInDrawnItems(portal);
+		} else {
+			return self.portalInScreen(portal);
+		}
+	}
+	self.genStr = function genStr(p) {
+		var href = 'https://www.ingress.com/intel?ll=' + p._latlng.lat + ',' + p._latlng.lng + '&z=17&pll=' + p._latlng.lat + ',' + p._latlng.lng;
+		var str = p.options.data.title.replace(/\"/g, "\\\"");
+		str = str.replace(';',' ');
+		if (window.plugin.keys) {
+			var keyCount = window.plugin.keys.keys[x] || 0;
+			str = str + ";" +href +";" + keyCount;
+		} else {
+			str = str + ";" + href;
+		}
+		return str;
+	}
 
-	self.gen = function gen() {
-			var o = [];
-			var antal = 0;
-			var tooMany = 0;
-
-			if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
-				var inBounds = function(portal) {
-					return self.portalInDrawnItems(portal);
-				}
-				var string = "Portal selection based on drawTools boundaries.";
-			} else {
-				var inBounds = function(portal) {
-					return self.portalInScreen(portal);
-				}
-				var string = "Portal selection based on screen boundaries.";
-			}
-
-			for (var x in window.portals) {
-				var p = window.portals[x];
-				if (antal < 50) {
-					if (inBounds(p)) {
-						antal = antal + 1;
-						var href = 'https://www.ingress.com/intel?ll=' + p._latlng.lat + ',' + p._latlng.lng + '&z=17&pll=' + p._latlng.lat + ',' + p._latlng.lng;
-						var str1 = p.options.data.title.replace(/\"/g, "\\\"");
-						var str2 = str1.replace(';', ' ');
-
-						//check for keys plugin
-						if (window.plugin.keys) {
-							var keyCount = window.plugin.keys.keys[x] || 0;
-							o.push(str2 + ";" + href + ";" + keyCount);
-						} else {
-							o.push(str2 + ";" + href);
-						}
-					}
+	self.checkPortals= function checkPortals(portals) {
+		var count=0;
+		var tooMany=false;
+		var list;
+		for (var x in portals) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/stefan-nygren/IITC-Ingress-Maxfields-Exporter/file/1580691871/issues/source?bid=2892513&fileBranchId=3023918#l96)